src/main/webapp/app/shared/util/core-drive-annotation-submission/core-drive-annotation-submission.ts

```diff
@@ -0,0 +1,136 @@
+import _ from 'lodash';
+import { Drug, Gene, Mutation, Review, Treatment, Tumor, Vus, DrugCollection } from '../../model/firebase/firebase.model';
+
+export function getGeneData(geneData: Gene, excludeComments: boolean, onlyReviewedContent: boolean, drugList: DrugCollection): true | Gene {
+  const gene = _.cloneDeep(geneData);
+  processData(gene, ['summary', 'background'], excludeComments, onlyReviewedContent);
+  processData(gene.type, ['tsg', 'ocg'], excludeComments, onlyReviewedContent);
+  const tempMutations: Mutation[] = [];
+  for (const mutation of gene.mutations ?? []) {
+    const tempTumors: Tumor[] = [];
+    if (shouldExclude(onlyReviewedContent, mutation.name_review)) {
+      tempMutations.push(mutation);
+      continue;
+    }
+    processData(mutation, ['name'], excludeComments, onlyReviewedContent);
+    processData(mutation.mutation_effect, ['oncogenic', 'effect', 'description'], excludeComments, onlyReviewedContent);
+    for (const tumor of mutation.tumors ?? []) {
+      if (shouldExclude(onlyReviewedContent, tumor.cancerTypes_review)) {
+        tempTumors.push(tumor);
+        continue;
+      }
+      // process tumor cancerTypes
+      processData(tumor, ['summary', 'diagnosticSummary', 'prognosticSummary'], excludeComments, onlyReviewedContent);
+      processData(tumor.diagnostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      processData(tumor.prognostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
```

Comment:
@bprize15  could you remind me whether we still use relevantCancerTypes or we started to use excludedCancerTypes?

Reply:
If I remember correctly, we are talking about how a curator might exclude a cancer type from a treatment?

If so, I believe we decided that a curator might used the RCT modal when they need to exclude many cancer types and that they would most likely exclude the cancer type directly in the name (using `excludedCancerTypes`). 

Since we support both ways, I imagine the logic would need to consider both methods of excluding cancer types. However, based on our conversations (among ourselves and with curators), I believe they most commonly use `excludedCancerTypes`.

Reply:
@zhx828 should I change `relevantCancerTypes` to `excludedCancerTypes`?

Reply:
@jfkonecn let's pass both in to the server side. We might have legacy code storing relevantCancerTypes.